karapace/kafka_rest_apis/admin.py

```diff
@@ -40,36 +40,23 @@ def get_topic_config(self, topic: str) -> dict:
     def new_topic(self, name: str):
         self.create_topics([NewTopic(name, 1, 1)])
 
-    def cluster_metadata(self, topics: List[str] = None, retries=0) -> dict:
+    def cluster_metadata(self, topics: List[str] = None, retries: int = 0) -> dict:
         """List all kafka topics."""
-        node_found = True
         metadata_version = self._matching_api_version(MetadataRequest)
         if metadata_version > 6 or metadata_version < 1:
             raise UnrecognizedBrokerVersion(
                 "Kafka Admin interface cannot determine the controller using MetadataRequest_v{}.".format(metadata_version)
             )
-        # update_metadata on cluster only works with v1 type metadata
         request = MetadataRequest[1](topics=topics)
-        with self.client_lock:
-            node_id = self._client.least_loaded_node()
-            # normal admin code will call poll on client not ready which will sometime block indefinitely on select
-            if not node_id or not self.first_boot or not self._client.ready(node_id):
-                #  pylint: disable=protected-access
-                node_id = list(self._client.cluster._bootstrap_brokers.keys())[0]
-                self.first_boot = False
```

Comment:
We could remove `first_boot` from `__init__` as well.

Reply:
nice catch, done !